# Generated by Django 4.2.3 on 2024-07-30 13:34

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('dogovornoy', '0011_delete_notification'),
    ]

    operations = [
        migrations.CreateModel(
            name='TaskReason',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('reason', models.CharField(max_length=255, verbose_name='Причина')),
            ],
            options={
                'verbose_name': 'Причина заявки',
                'verbose_name_plural': 'Причины заявок',
            },
        ),
        migrations.CreateModel(
            name='TechnicalTask',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('client_object_id', models.IntegerField()),
                ('sent_time', models.DateTimeField(auto_now_add=True)),
                ('accepted_time', models.DateTimeField(blank=True, null=True)),
                ('arrival_time', models.DateTimeField(blank=True, null=True)),
                ('completion_time', models.DateTimeField(blank=True, null=True)),
                ('note', models.TextField(blank=True)),
                ('result', models.TextField(blank=True)),
                ('reason', models.ForeignKey(null=True, on_delete=django.db.models.deletion.SET_NULL, to='dogovornoy.taskreason')),
                ('sender', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='sent_tasks', to=settings.AUTH_USER_MODEL)),
                ('technician', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='received_tasks', to=settings.AUTH_USER_MODEL)),
            ],
        ),
    ]
